In [123]:
import pandas as pd
data=pd.read_csv("../datasets/5.-Economics-&-Business/unemployment.csv")

In [124]:
dict={'January':1,'February':2,'March':3,'April':4,'May':5,'June':6,'July':7,'August':8,'September':9,'October':10,'November':11,'December':12}

In [125]:
pd.set_option('display.max_rows', 500)

# Do we have more male of female unemployed?

In [234]:
data_gender = data.groupby(['Year','Month','Gender']).sum ()
data_gender.rename(dict,inplace=True)
data_gender.sort_values(['Year','Month'],ascending = True,inplace=True)
data_gender
data_gender.drop([ 'District Code', 'Neighborhood Code'], axis=1,inplace=True)
data_gender

Number
Year Month Gender        
2013 1     Female   56175
           Male     58268
     2     Female   56352
           Male     58584
     3     Female   56490
           Male     58664
     4     Female   55911
           Male     57959
     5     Female   55007
           Male     56444
     6     Female   53560
           Male     54353
     7     Female   54358
           Male     53834
     8     Female   54789
           Male     53817
     9     Female   54587
           Male     54119
     10    Female   54952
           Male     55032
     11    Female   54920
           Male     55361
     12    Female   53451
           Male     54226
2014 1     Female   54183
           Male     54968
     2     Female   53780
           Male     54445
     3     Female   53578
           Male     54156
     4     Female   52928
           Male     53470
     5     Female   51452
           Male     51332
     6     Female   50835
           Male     50108
     7     Female   52091
           Male     49906
     8     Female   52394
           Male     49702
     9     Female   51503
           Male     49542
     10    Female   51437
           Male     50122
     11    Female   50909
           Male     49694
     12    Female   50024
           Male     49032
2015 1     Female   61668
           Male     61747
     2     Female   61945
           Male     61489
     3     Female   61510
           Male     60752
     4     Female   60277
           Male     59020
     5     Female   58748
           Male     56657
     6     Female   58456
           Male     54684
     7     Female   60205
           Male     53396
     8     Female   61094
           Male     53563
     9     Female   59243
           Male     53401
     10    Female   58270
           Male     54087
     11    Female   58320
           Male     54471
     12    Female   56985
           Male     53750
2016 1     Female   57223
           Male     53778
     2     Female   56511
           Male     52936
     3     Female   56171
           Male     52468
     4     Female   54846
           Male     50941
     5     Female   53491
           Male     49366
     6     Female   53793
           Male     47677
     7     Female   55147
           Male     46529
     8     Female   56041
           Male     46530
     9     Female   54199
           Male     46499
     10    Female   52800
           Male     46676
     11    Female   52844
           Male     47533
     12    Female   51488
           Male     46597
2017 1     Female   51680
           Male     46886
     2     Female   52374
           Male     47171
     3     Female   52307
           Male     46752
     4     Female   51034
           Male     45006
     5     Female   50184
           Male     43645
     6     Female   50000
           Male     42132
     7     Female   51375
           Male     41367
     8     Female   52388
           Male     41775
     9     Female   50579
           Male     41516
     10    Female   50108
           Male     42403
     11    Female   50867
           Male     43830
     12    Female   49569
           Male     43161

In [236]:
data_gender.groupby(level=['Year', 'Gender']).sum()

Number
Year Gender        
2013 Female  660552
     Male    670661
2014 Female  625114
     Male    616477
2015 Female  716721
     Male    677017
2016 Female  654554
     Male    587530
2017 Female  612465
     Male    525644

In [252]:
data_gender.loc[2013,'Number'].sum()# unemployment in 2013

1331213

In [253]:
data_gender.loc[2014,'Number'].sum()# unemployment in 2014

1241591

In [254]:
data_gender.loc[2015,'Number'].sum()# unemployment in 2015

1393738

In [255]:
data_gender.loc[2016,'Number'].sum()# unemployment in 2016

1242084

In [256]:
data_gender.loc[2017,'Number'].sum()# unemployment in 2017

1138109

In [261]:
#Finding the sum of each year
data_max=data_gender.groupby(['Year']).sum()
data_max

Number
Year         
2013  1331213
2014  1241591
2015  1393738
2016  1242084
2017  1138109

In [265]:
data_max.loc[:,'Number'].max()

1393738

# On average which is the month of the year where we register the most unemployment?

In [194]:
data_per_month=data_gender.groupby(['Month','Gender']).sum()
data_per_month

Number
Month Gender        
1     Female  280929
      Male    275647
2     Female  280962
      Male    274625
3     Female  280056
      Male    272792
4     Female  274996
      Male    266396
5     Female  268882
      Male    257444
6     Female  266644
      Male    248954
7     Female  273176
      Male    245032
8     Female  276706
      Male    245387
9     Female  270111
      Male    245077
10    Female  267567
      Male    248320
11    Female  267860
      Male    250889
12    Female  261517
      Male    246766

Female yearly evolution

In [207]:
data_per_month.loc[(1,'Female')]#Data in January

Number    280929
Name: (1, Female), dtype: int64

In [206]:
data_per_month.loc[(5,'Female')]#Data in May

Number    268882
Name: (5, Female), dtype: int64

In [205]:
data_per_month.loc[(9,'Female')]#Data in September

Number    270111
Name: (9, Female), dtype: int64

In [204]:
data_per_month.loc[(12,'Female')]#Data in December

Number    261517
Name: (12, Female), dtype: int64

In [214]:
#we can observe that  at the beginning of the year the unemployment rate is the highest

Male yearly evolution

In [210]:
data_per_month.loc[(1,'Male')]#Data in January

Number    275647
Name: (1, Male), dtype: int64

In [211]:
data_per_month.loc[(5,'Male')]#Data in May

Number    257444
Name: (5, Male), dtype: int64

In [212]:
data_per_month.loc[(9,'Male')]#Data in September

Number    245077
Name: (9, Male), dtype: int64

In [213]:
data_per_month.loc[(12,'Male')]#Data in December

Number    246766
Name: (12, Male), dtype: int64

In [215]:
# we can observe that the unemployment rate is the highest at the beginning of the year.

# What is the evolution of unemployment since 2013/2017?

In [178]:
data_total = data.groupby(['Year']).sum()
data_total.drop([ 'District Code', 'Neighborhood Code'], axis=1,inplace=True)
data_total

Number
Year         
2013  1331213
2014  1241591
2015  1393738
2016  1242084
2017  1138109

In [180]:
y_2013=data_total.iloc[0,0]

In [181]:
y_2017=data.iloc[4,4]

In [183]:
(y_2017)-(y_2013) # This means that we have less unemployment in 2017 than in 2013.

-1331208

In [216]:
#Do we have more male of female unemployed?

#data_male_2017=data.loc[(data.Gender == 'Male') & (data.Year ==2017),]
#data_male_2017
#data_male_2017.loc[:,'Number'].sum()
#so we have 525644 male unemployed in 2017
#data_male_2017_january=data.loc[(data.Gender == 'Male') & (data.Year ==2017) & (data.Month == 'January'),]
#data_male_2017_january.loc[:,'Number'].sum()
#data_male_2017_may=data.loc[(data.Gender == 'Male') & (data.Year ==2017) & (data.Month == 'May'),]
#data_male_2017_may.loc[:,'Number'].sum()
#data_male_2017_sept=data.loc[(data.Gender == 'Male') & (data.Year ==2017) & (data.Month == 'September'),]
#data_male_2017_sept.loc[:,'Number'].sum()
#data_male_2017_dec=data.loc[(data.Gender == 'Male') & (data.Year ==2017) & (data.Month == 'December'),]
#data_male_2017_dec.loc[:,'Number'].sum()


#data_female_2017=data.loc[(data.Gender == 'Female') & (data.Year ==2017),]
#data_female_2017.loc[:,'Number'].sum()
#data_female_2017_january=data.loc[(data.Gender == 'Female') & (data.Year ==2017) & (data.Month == 'January'),]
#data_female_2017_january.loc[:,'Number'].sum()
#data_female_2017_may= data.loc[(data.Gender == 'Female') & (data.Year == 2017) & (data.Month == 'May'),]
#data_female_2017_may.loc[:,'Number'].sum()
#data_female_2017_sept= data.loc[(data.Gender == 'Female') & (data.Year == 2017) & (data.Month == 'September'),]
#data_female_2017_sept.loc[:,'Number'].sum()
#data_female_2017_dec= data.loc[(data.Gender == 'Female') & (data.Year == 2017) & (data.Month == 'December'),]
#data_female_2017_dec.loc[:,'Number'].sum()


####data_male_2013.groupby(['District Code', 'Month']).sum()
####temp_df.columns = [2017]
#####temp_df[2018] = other_df['Number']

We have 525644 male unemployed in 2017:

--The unemploymentin January is: 46886
--The unemployment in May is: 43645
--The unemployed male in September is 41516
--The number of unemployed male in December is 43161

We have 612465 female unemployed in 2017:

--The number of female unemplyed in January 2017 is 51680
--The number of female unemployed in May 2017 is 50184 
--The number of female unemployed in September 2017 is 50579
--The number of female unemployed in December 2017 is 49569